webtau-http/src/main/java/org/testingisdocumenting/webtau/http/datanode/DataNode.java

```diff
@@ -21,35 +22,59 @@
 import org.testingisdocumenting.webtau.expectation.equality.CompareToComparator;
 import org.testingisdocumenting.webtau.expectation.equality.CompareToResult;
 
+import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 
 import static org.testingisdocumenting.webtau.WebTauCore.createActualPath;
 
-public interface DataNode extends DataNodeExpectations, Comparable, Iterable<DataNode> {
+public interface DataNode extends DataNodeExpectations, Comparable<Object>, Iterable<DataNode> {
     DataNodeId id();
 
     DataNode get(String pathOrName);
 
-    boolean has(String pathOrName);
+    default boolean has(String pathOrName) {
+        return !get(pathOrName).isNull();
+    }
 
     DataNode get(int idx);
 
-    TraceableValue getTraceableValue();
+    default TraceableValue getTraceableValue() {
+        return null;
+    }
 
     <E> E get();
 
-    boolean isList();
+    default boolean isList() {
+        return false;
+    }
 
-    boolean isSingleValue();
+    default boolean isSingleValue() {
+        return false;
+    }
 
-    List<DataNode> elements();
+    default List<DataNode> elements() {
+        return Collections.emptyList();
+    }
 
-    int numberOfChildren();
+    @Override
+    @SuppressWarnings("NullableProblems")
+    default Iterator<DataNode> iterator() {
+        return elements().iterator();
+    }
+
+    default int numberOfChildren() {
+        return 0;
+    }
 
-    int numberOfElements();
+    default int numberOfElements() {
+        return 0;
+    }
 
-    Map<String, DataNode> asMap();
+    default Map<String, DataNode> asMap() {
```

Comment:
have mixed feelings about these defaults. It feels like it is not really a default implementation, but more like you know 2 out of 3 implementations are similar and hence they are there. Maybe it is OK in these case since we are not going to have more implementations.

Reply:
Yes, I was equally in two minds.  I did it not so much for removing duplication but rather for decluttering the specific impls.  For example, the list impl only worries about functions relevant to list, not all this other stuff.

Reply:
In a unlikely event that someone will implement DataNode interface I would only left real defaults here.

Reply:
There was more I initially had as defaults but held back a bit.  Which ones did you have in mind?